lib/phoenix_component.ex

```diff
@@ -1025,13 +1233,142 @@ defmodule Phoenix.Component do
     attrs
   end
 
-  defp raise_if_function_already_defined!(env, name, attrs) do
+  defp get_slots(module) do
+    Module.get_attribute(module, :__slots__) || []
+  end
+
+  defp pop_slots(env) do
+    slots =
+      env.module
+      |> get_slots()
+      |> Enum.reverse()
+
+    Module.delete_attribute(env.module, :__slots__)
```

Comment:
I think we don't need to define slots as a separate attribute, we can put them inside `__attrs__` themselves. WDYT? This should reduce the amount of code duplication between handling `__slots__` and `__attrs__`.

Reply:
🤔 At the moment I prefer keeping them separate, as slots can have children (slot attributes and inner content), but attributes cannot, so they feel like distinct concerns.

I don't have a strong opinion either way, so let me know if I should get started on this next 🙂 Alternatively, could this kind of refactor could be a follow-up PR once we get initial feedback from users?

Reply:
Agreed! Resolving this...